ocs_ci/ocs/scale_lib.py

```diff
@@ -0,0 +1,385 @@
+        )
+        # Appending all the pvc_obj and pod_obj to list
+        pvc_objs, pod_objs = ([] for i in range(2))
+        pvc_objs.extend(cephfs_pvcs + rbd_pvcs)
+
+        # Create pods with above pvc list
+        cephfs_pods = helpers.create_pods_parallel(
+            cephfs_pvcs, self.namespace, constants.CEPHFS_INTERFACE,
+            pod_dict_path=str(self.pod_dict_path), sa_name=self.sa_name.name,
+            dc_deployment=self.dc_deployment, node_selector=self.node_selector
+        )
+        rbd_rwo_pvc, rbd_rwx_pvc = ([] for i in range(2))
+        for pvc_obj in rbd_pvcs:
+            if pvc_obj.get_pvc_access_mode == constants.ACCESS_MODE_RWX:
+                rbd_rwx_pvc.append(pvc_obj)
+            else:
+                rbd_rwo_pvc.append(pvc_obj)
+        rbd_rwo_pods = helpers.create_pods_parallel(
+            rbd_rwo_pvc, self.namespace, constants.CEPHBLOCKPOOL,
+            pod_dict_path=str(self.pod_dict_path), sa_name=self.sa_name.name,
+            dc_deployment=self.dc_deployment, node_selector=self.node_selector
+        )
+        rbd_rwx_pods = helpers.create_pods_parallel(
+            rbd_rwx_pvc, self.namespace, constants.CEPHBLOCKPOOL,
+            pod_dict_path=str(self.pod_dict_path), sa_name=self.sa_name.name,
+            dc_deployment=self.dc_deployment, raw_block_pv=True,
+            node_selector=self.node_selector
+        )
+        temp_pod_objs = list()
+        temp_pod_objs.extend(cephfs_pods + rbd_rwo_pods)
+
+        # Appending all the pod_obj to list
+        pod_objs.extend(temp_pod_objs + rbd_rwx_pods)
+
+        fio_size = self.get_size_based_on_cls_usage()
+        fio_rate = self.get_rate_based_on_cls_iops()
+
+        # Start IO
+        if start_io:
+            threads = list()
+            for pod_obj in temp_pod_objs:
+                process = threading.Thread(
+                    target=pod_obj.run_io, kwargs={
+                        'storage_type': 'fs', 'size': fio_size,
+                        'runtime': io_runtime, 'rate': fio_rate
+                    }
+                )
+                process.start()
+                threads.append(process)
+            for pod_obj in rbd_rwx_pods:
+                process = threading.Thread(
+                    target=pod_obj.run_io, kwargs={
+                        'storage_type': 'block', 'size': fio_size,
+                        'runtime': io_runtime, 'rate': fio_rate
+                    }
+                )
+                process.start()
+                threads.append(process)
+            for process in threads:
+                process.join()
+
+        return pod_objs, pvc_objs
+
+    def get_size_based_on_cls_usage(self, custom_size_dict=None):
+        """
+        Function to check cls capacity suggest IO write to cluster
+        Args:
+            custom_size_dict (dict): Dictionary of size param to be used during IO run.
```

Comment:
What is the significance of `custom_size_dict` if this function is to get the size based on cluster usage? it kills the purpose, isn't it?

Reply:
The purpose is to change the fio size param according to the cluster utilization, so after each iteration this function will be called and according to the cluster utilization fio_size param will be changed. 
Without this, all the pods will have io will same size and problem is cluster capacity and utilization. If we have some constant value then depends on 2/4/0.5 TiB cluster there will be problem like lesser or over utilization.